trace/trace.go

```diff
@@ -145,13 +154,25 @@ func WithSampler(sampler Sampler) StartOption {
 	}
 }
 
+// WithBufferLimit makes new spans with a custom buffer limit.
+// This limits variable trace span data: Message Events, Links, and Annotations.
+// The default is 1000 (trace.DefaultBufferLimit) and the minimum is 1.
+func WithBufferLimit(limit int) StartOption {
+	return func(o *StartOptions) {
+		o.BufferLimit = limit
+		if o.BufferLimit <= 0 {
+			o.BufferLimit = 1
```

Comment:
it should be the default not 1.

Reply:
We've decided on a different strategy (per @ramonza). The client can pass anything in. If it's <= 0, then that trace type is disabled. I think it simplifies things.